.github/workflows/codeql.yml

```diff
@@ -0,0 +1,52 @@
+name: "CodeQL"
```

Comment:
should have a comment block at the top explaining what this is about and links to relevant docs